docs/understanding-preprocessing-codes.md

```diff
@@ -0,0 +1,490 @@
+
+```R
+enhancement_stop <- enhancement
+
+enhancement_stop$Origen_pa <- "Bosque Inicial"
+
+enhancement_stop$year <- year_t2+1
+
+enhancement <- rbind(enhancement,enhancement_stop)
+
+enhancement<-st_transform(enhancement, "+proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0")
+
+enhancement$year<-as.integer(enhancement$year)
+
+write_sf(enhancement, paste0(output_gcbm, "/forest_enhancement_LosRios.shp"))
+```
+
+In a similar manner, we are going to modify the growth curves to reflect the enhancements and create the growth curves including all the forest enhancement curves. A loop that starts with the enhancement outside conservation areas and then with the enhancement inside degradation areas is initiated.
+
+```R
+for (i in 1:(n_quantiles*2)) {
+  if (i<= n_quantiles){
+    enhancement_name <- paste0("Forest ", "Enhancement Chile"," intensity lvl ",i)
+  } else {
+    enhancement_name <- paste0("Forest ", "Conservation Enhancement Chile"," intensity lvl ",i-20)
+  }
+
+  growth_base <- filter(growth,Origen %in% c("Bosque Inicial"))
+  growth_base$Origen <- enhancement_name
+  if (i<= n_quantiles) {
+    year_grow <- unname(means_quantile$mean_quan[i])
+  } else {
+    year_grow <- unname(means_quantile_c$mean_quan[i-n_quantiles])
+  }
+  year_grow_vol <- year_grow / (1.75 * 0.5 * 0.5 * (44/12))
+  growth_curve <- seq(from= 0, by=year_grow_vol,length.out = ncol(growth_base)-4)
+  growth_curve <- unname(rbind(growth_curve))
+  growth_curve <- growth_curve[rep(seq_len(nrow(growth_curve)), nrow(growth_base)), ]
+  growth_base[,5:ncol(growth_base)] <- growth_curve
+  if (i==1){
+    growth_full <- rbind(growth, growth_base)
+  } else {
+    growth_full <- rbind(growth_full,growth_base)
+  }
+}
+```
+
+In a similar manner to create the growth curves for "No forestal" (non-forest) regions, a loop is initiated that starts with the enhancement outside conservation areas and then with the enhancement inside degradation areas.
+
+```R
+for (i in 1:(n_quantiles*2)) {
+  if (i<= n_quantiles){
+    enhancement_name <- paste0("Forest ", "Enhancement Chile"," intensity lvl ",i)
+  } else {
+    enhancement_name <- paste0("Forest ", "Conservation Enhancement Chile"," intensity lvl ",i-20)
+  }
+
+  growth_curve <- c("No forestal","No forestal",enhancement_name,"Not stocked",rep(as.numeric(0), ncol(growth_base)-4))
+  growth_curve
+  growth_full <- rbind(growth_full,growth_curve)
+}
+```
+
+Finally, we use the "non stocked" species so that the GCBM will ignore the growth and append it to the dataframe. The volume values are converted to numeric data and classifiers are converted to text. We finally write the CSV with the growth curves including the enhancement curves.
```

Comment:
put a comma after "ignore the growth" and specify what is appended into the dataframe (the non-stocked growth curves)

Reply:
Done!